client/shaders/nodes_shader/opengl_fragment.glsl

```diff
@@ -381,6 +384,8 @@ void main(void)
 	color = base.rgb;
 	vec4 col = vec4(color.rgb * varColor.rgb, 1.0);
 
+	col.rgb += ambientLight * ambientColor.rgb;
```

Comment:
The ambient light needs to be multiplied by the diffuse colour, i.e. `base.rgb`. Otherwise, it's not an ambient light (as defined in the Phong model) but looks more like something foggy.

Side note: To be precise, the addition is wrong/inaccurate because colours are in the non-linear sRGB colour space, but this is another issue which cannot be fixed in this Pull Request.

Reply:
Didn't think about that. The even more correct way is It should be added with `varColor.rgb` and then multiplied by 'color.rgb`: `vec4 col = vec4(color.rgb * (varColor.rgb + ambientLight * ambientColor.rgb));`. Since `varColor.rgb` is a diffuse light having the sun and artificial light combined, `ambientLight` is some constant light, those components should be added in the Phong model.

Reply:
Fixed. Now the ambient light is added with `varColor` (in the shaders) or with the light color calculated in `final_color_blend()` (in the case if the shaders are disabled).